### PR TITLE
qutebrowser: 1.14.1 -> 2.0.1

### PR DESCRIPTION
**Pull request by ghost**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/adblock/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/adblock/default.nix:47]: Please move this into preInstallPhase.

File: pkgs/applications/networking/browsers/qutebrowser/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/qutebrowser/default.nix:73]: Please use pythonOlderThan "3.9" to do this automatically.

File: pkgs/applications/networking/browsers/qutebrowser/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/qutebrowser/default.nix:40]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/browsers/qutebrowser/default.nix
Comment (petabyteboy) [reply pkgs/applications/networking/browsers/qutebrowser/default.nix:40]: This is a draft PR. This is commented out because I need to temporarily use the sources from git and I will change it back to fetchurl once release tarballs exist (which means before this is merged).

File: pkgs/development/python-modules/adblock/default.nix
Comment (petabyteboy) [reply pkgs/development/python-modules/adblock/default.nix:47]: This is actually part of the diff for https://github.com/NixOS/nixpkgs/pull/108271. And there is no such thing as `preInstallPhase`. I could move it to preInstall, but that would make it harder to use overrideAttrs to add a new preInstall. So I think it's good like this. It's also copied from pkgs/development/python-modules/retworkx/default.nix. If you still think this should be different, please comment on https://github.com/NixOS/nixpkgs/pull/108271.

File: pkgs/development/python-modules/adblock/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/adblock/default.nix:61]:
```suggestion
```

set by buildRustPackage

Comment (petabyteboy) [reply pkgs/development/python-modules/adblock/default.nix:61]: The change you commented on is already merged, see https://github.com/NixOS/nixpkgs/pull/108271

**Comment by phenax** [comment]
They've created a release - https://github.com/qutebrowser/qutebrowser/releases/tag/v2.0.0

File: pkgs/applications/networking/browsers/qutebrowser/default.nix
Comment (DIzFer) [inline pkgs/applications/networking/browsers/qutebrowser/default.nix:79]: Testing in my excitement revealed nixos-unstable doesn't like this line.

pythonOlderThan seems to exist as pythonOlder, and i had to surround it in parenthesis for it to evaluate.

Does that sound right?

Comment (rnhmjoj) [reply pkgs/applications/networking/browsers/qutebrowser/default.nix:79]: Yes, the parenthesis are certainly missing.

Comment (SuperSandro2000) [reply pkgs/applications/networking/browsers/qutebrowser/default.nix:79]:
```suggestion
    ++ lib.optional (pythonOlderThan "3.9") importlib-resources
```

**Comment by The-Compiler** [comment]
Upstream here - FWIW I'll probably follow up with a v2.0.1 release soon (possibly even today), since [a couple of issues](https://github.com/qutebrowser/qutebrowser/blob/master/doc/changelog.asciidoc#v2.0.1) were reported shortly after the release (like it happens usually... :laughing:).

File: pkgs/applications/networking/browsers/qutebrowser/default.nix
Comment (dotlambda) [inline pkgs/applications/networking/browsers/qutebrowser/default.nix:60]:
```suggestion
    pyopengl
```
https://github.com/qutebrowser/qutebrowser/blob/master/doc/changelog.asciidoc#v200-2021-01-28

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/browsers/qutebrowser/default.nix
Comment (The-Compiler) [reply pkgs/applications/networking/browsers/qutebrowser/default.nix:60]: PyOpenGL isn't needed either - that dependency got [removed in v1.0.0](https://github.com/qutebrowser/qutebrowser/blob/master/doc/changelog.asciidoc#v100-2017-10-12) back in 2017 :wink:

Comment (petabyteboy) [reply pkgs/applications/networking/browsers/qutebrowser/default.nix]: Thanks for the hints!

File: pkgs/applications/networking/browsers/qutebrowser/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/qutebrowser/default.nix:66]:
```suggestion
  ] ++ lib.optional (pythonOlder "3.9") importlib-resources
```

Comment (petabyteboy) [reply pkgs/applications/networking/browsers/qutebrowser/default.nix:66]: Why should I change this?

**Review by dotlambda — APPROVED** [review]

**Comment by The-Compiler** [comment]
Yay! :tada: Note that a v2.0.2 isn't too far away since there are still some [regressions to deal with](https://github.com/qutebrowser/qutebrowser/issues?q=is%3Aissue+milestone%3Av2.0.2) - currently aiming for next week, but might be earlier if I can find the time.

**Comment by dotlambda** [comment]
Thank you so much @The-Compiler for even having an eye on what distributions are doing w.r.t. qutebrowser. That's so much more than what the average upstream maintainer does!

**Pull request merged**
